metadata/images.go

```diff
@@ -55,6 +55,36 @@ func (s *imageStore) Get(ctx context.Context, name string) (images.Image, error)
 
 	if err := view(ctx, s.db, func(tx *bolt.Tx) error {
 		bkt := getImagesBucket(tx, namespace)
+		if bkt == nil {
+			nsbkt := getNamespacesBucket(tx)
+			cur := nsbkt.Cursor()
+			for k, _ := cur.First(); k != nil; k, _ = cur.Next() {
+				// Check if v has shared values
+				labelsBkt := getNamespaceLabelsBucket(tx, string(k))
+				if labelsBkt == nil {
+					continue
+				}
+				labels := map[string]string{}
+				labelsBkt.ForEach(func(k, v []byte) error {
+					labels[string(k)] = string(v)
+					return nil
+				})
+				// if it is, check if K has image.
+				if _, ok := labels[sharedNamespaceLabel]; ok {
```

Comment:
Same as another loop. We don't have to collect all labels to see the existence of sharedNamespaceLabel.

Reply:
We also only care about the `sharedNamespaceLabel` key being present; there's no benefit to storing any of the other keys or any of the values.